docs/RFCS/20200401_alter_column_type.md

```diff
@@ -0,0 +1,87 @@
+1. Create a new hidden computed column c' using c::t'
+2. Make a copy of the primary key pk and replace each reference to c with c'
+    - pk' must be encoded as a primary index, cannot switch secondary index to primary index
+3. Each index i that references pk, create an index i' that references pk'
+4. Enqueue primary index swap with pk and pk', and index swaps with i and i'. 
+5. Swap c' with c. 
+6. Enqueue drops for the old column and old indexes including the primary key.
+
+
+
+# Problems
+## Out of order ColumnIDs/OrdinalPositions/attnum
+When performing a column swap, ColumnIDs cannot be swapped due to encodings. For the current listed method of performing a swap by adding a new column and dropping the old one, the ColumnIDs will be "out of order". This results in places that currently depend on ColumnIDs to also be out of order. For example in information_schema.columns, ordinal_position is ColumnID. (Similarly attnum and adnum in pg_catalog also correpsond to ColumnID).
+This results in SHOW COLUMNS being out of order as it currently depends on column id. 
+
+One solution is adding a "ordinal position" slice field to the TableDescriptor to handle the "logical id" of a column for virtual tables. As it currently stands, this ordinal position value can correspond to the Column ID, it would only change when doing this column swap, so when a swap is performed, the ordinal positions of the columns change. We can store ordinal position in the ColumnDescriptor or TableDescriptor. If we use a slice for OrdinalPositions in table descriptor, each index can correspond to the Columns field in TableDescriptor, thus when doing a swap, between two columns, we wouldn't have to change the OrdinalPositions.
+
+Interesting page on ALTER COLUMN POSITION idea for Postgres.
+https://wiki.postgresql.org/wiki/Alter_column_position
+
+This blurb mentions using separate logical and physical identifiers for the column.
+```
+Adding alter column syntax into postgres
+Since the above methods have a number of issues, it has often been expressed that we would like to add capabilities for postgres to allow reordering of columns. The flip side of this is that it would also be desirable for postgres to automatically order columns physically for optimum layout, regardless of the logical order they are given in. The current problem with implementing this lies in that currently postgres uses the same identifiers for both the logical and physical position within a table. The current hot plan for solving this would be to change the system to reference three identifiers... a permanent identifier for the column, as well as a separate logical and physical identifier. This would allow places that need to deal specifically with column order at the logical level (ie. select *) to reference the logical number, while places that interact with disk system can access the physical number, and all other places just use the column's permanent id.
```

Comment:
I'm not sure you need to include the entire blurb here -- including a link and maybe pulling out an important part or two is good for the RFC

Reply:
Removed